packages/components/organisms/f-checkout/src/components/_tests/AgeVerification.test.js

```diff
@@ -0,0 +1,231 @@
+                            customer: {
+                                dateOfBirth: null
+                            }
+                        })
+                    });
+
+                    // Act
+                    wrapper.vm.initDateOfBirth();
+                });
+
+                it('should update `selectedDate.day` with "1"', () => {
+                    // Assert
+                    expect(wrapper.vm.selectedDate.day).toEqual('1');
+                });
+
+                it('should update `selectedDate.month` with "0"', () => {
+                    // Assert
+                    expect(wrapper.vm.selectedDate.month).toEqual('0');
+                });
+
+                it('should update `selectedDate.year` with the current year', () => {
+                    // Assert
+                    // expect(dateNowSpy).toHaveBeenCalled();
+                    expect(wrapper.vm.selectedDate.year).toEqual(currentYear);
+                });
+            });
+        });
+
+        describe('`selectionChanged` ::', () => {
+            it.each([
+                ['day', '7'],
+                ['month', '4'],
+                ['year', '1990']
+            ])('should update `selectedDate[%s]` with %s`', (type, selection) => {
+                // Arrange
+                wrapper = shallowMount(AgeVerification, {
+                    i18n,
+                    localVue,
+                    store: createStore({
+                        ...defaultCheckoutState,
+                        customer: {
+                            dateOfBirth: new Date('2010', '9', '2')
+                        }
+                    })
+                });
+
+                // Act
+                wrapper.vm.selectionChanged(selection, type);
+
+                // Assert
+                expect(wrapper.vm.selectedDate[type]).toEqual(selection);
+            });
+        });
+
+        describe('`handleAgeVerifcation` ::', () => {
+            // Arrange
+            let selectedDate;
+            let date;
+            let updateDateOfBirthSpy;
+
+            beforeEach(() => {
+                // Arrange
+                selectedDate = {
+                    day: '1',
+                    month: '7',
+                    year: '1987'
+                };
+
+                date = new Date(selectedDate.year, selectedDate.month, selectedDate.day);
+                updateDateOfBirthSpy = jest.spyOn(AgeVerification.methods, 'updateDateOfBirth');
+
+                wrapper = shallowMount(AgeVerification, {
+                    i18n,
+                    localVue,
+                    store: createStore({
+                        ...defaultCheckoutState,
+                        customer: {
+                            dateOfBirth: new Date('2010', '9', '2')
+                        }
+                    })
+                });
+            });
+
+            it('should call `updateDateOfBirth` with `selectedDate`', () => {
```

Comment:
I'm not a test guru by any means, but rather than testing that it calls `updateDateofBirth` (which you can do too), would it not be better to test that the state itself has been updated to the new date? (as that's the desired result)

Reply:
I can do, but `handleAgeVerifcation` just calls the action in the store that then calls the mutation that updates the state 🤯 .  So I thought you wouldn't test that the state changes because `handleAgeVerifcation` doesn't directly change the state?